node/cn/handler.go

```diff
@@ -1083,80 +1081,79 @@ func (pm *ProtocolManager) broadcastCNTx(txs types.Transactions) {
 		half := (len(peers) / 2) + 2
 		peers = samplingPeers(peers, half)
 		for _, peer := range peers {
-			txset[peer] = append(txset[peer], tx)
+			cnPeersWithoutTxs[peer] = append(cnPeersWithoutTxs[peer], tx)
 		}
 		logger.Trace("Broadcast transaction", "hash", tx.Hash(), "recipients", len(peers))
 	}
 
-	propTxPeersGauge.Update(int64(len(txset)))
+	propTxPeersGauge.Update(int64(len(cnPeersWithoutTxs)))
 	// FIXME include this again: peers = peers[:int(math.Sqrt(float64(len(peers))))]
-	for peer, txs := range txset {
+	for peer, txs := range cnPeersWithoutTxs {
 		//peer.SendTransactions(txs)
 		peer.AsyncSendTransactions(txs)
 	}
 }
 
-func (pm *ProtocolManager) broadcastNoCNTx(txs types.Transactions, resend bool) {
-	var cntxset = make(map[Peer]types.Transactions)
-	var txset = make(map[Peer]types.Transactions)
+func (pm *ProtocolManager) broadcastTxsFromPN(txs types.Transactions) {
+	cnPeersWithoutTxs := make(map[Peer]types.Transactions)
```

Comment:
For `samplingPeers`, Did you separate `cnPeersWithoutTxs` and `peersWithoutTxs`?

Reply:
There's nothing changed. `samplingPeers` was and is executed only for `cnPeersWithoutTxs`, not `peersWithoutTxs`.